admin-dev/themes/new-theme/js/product-preferences-page/index.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import TranslatableInput from '../components/translatable-input';
+import StockManagementOptionHandler from './stock-management-option-handler';
+
+const $ = window.$;
+
+$(() => {
```

Comment:
Sounds like this is specific to _product_preferences_ page, why not create a new Webpack entry named product_preference_page and include it into this page instead?

Reply:
It sounds like a good idea. I have added new entry in webpack, but still im not sure if it is the best option. if we add new entry for every page, webpack config may get really big, but maybe it's not a problem. :)

Reply:
I also wonder if we should create separate directory like 'pages' instead of suffixing each directory with '-page'. What do you think?